pkg/api/projectedresource/v1alpha1/share_type.go

```diff
@@ -0,0 +1,63 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Share is the Schema for the shares API
+type Share struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ShareSpec   `json:"spec,omitempty"`
+	Status ShareStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ShareList contains a list of Share
+type ShareList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Share `json:"items"`
+}
+
+// ShareSpec defines the desired state of Share
+type ShareSpec struct {
+	// BackingResource captures the ConfigMap or Secret that is shared.
+	BackingResource `json:"backingResource,omitempty"`
```

Comment:
This should be required (i.e. no omitempty).

Reply:
yep